### PR TITLE
Performance benchmarks

### PR DESCRIPTION
**Pull request by 0mp**
See the following for usage information:
 - `./run_benchmark --help`
 - https://github.com/apaszke/pytorch-dist/wiki/Benchmarking

### CODE REVIEW

File: torch/lib/THD/benchmark/benchmark.py [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+            print_stats(bytes, num_tensors, end - start)
+    print()
+elif rank == 1:
+    for bytes in [2**n for n in range(MIN_BYTES, MAX_BYTES)]:
+        tensor = torch.ByteTensor(bytes).fill_(42)
+        for num_tensors in [10**n for n in range(MIN_NUM_TENSORS, MAX_NUM_TENSORS)]:
+            for i in range(0, num_tensors):
+                dist.recv(tensor, 0)
+else:
+    pass
```
Comment (jytug) [inline torch/lib/THD/benchmark/benchmark.py]: Is there something to be implemented here?

File: torch/lib/THD/benchmark/run_benchmark [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+
+    -p, --port PORT
+        Set the port number master is listening on. Default: '29500'.
+
+    -s, --world-size WORLD_SIZE
+        Set the number of processes to be spawned. Default: '2'.
+EOF
+}
+
+while [ $# -gt 0 ]; do
```
Comment (jytug) [inline torch/lib/THD/benchmark/run_benchmark:76]: Shouldn't you check the parity of `$#` somewhere?

File: torch/lib/THD/benchmark/run_benchmark [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+            shift 2
+            ;;
+        *)
+            errxit "Unknown option '$1'"
+            ;;
+    esac
+done
+
+MASTER_ADDR="$master_hostname:$MASTER_PORT"
+if [ x"$BACKEND" = xtcp ]; then
```
Comment (jytug) [inline torch/lib/THD/benchmark/run_benchmark:137]: Nice

**Review by jytug — APPROVED** [review]
Looks good - most of the Python script is the same code repeated a few times, but it's your call whether or not you want to change it

File: torch/lib/THD/benchmark/benchmark.py (hunk 1 above)
Comment (0mp) [reply torch/lib/THD/benchmark/benchmark.py]: Nope. Only rank 0 and rank 1 should run this section.

Comment (jytug) [reply torch/lib/THD/benchmark/benchmark.py]: No need for the `pass` then

File: torch/lib/THD/benchmark/run_benchmark (hunk 2 above)
Comment (0mp) [reply torch/lib/THD/benchmark/run_benchmark:76]: `set -u` is on during the whole script so I guess I am not going to add any additional check. Good point though!

**Review by apaszke — APPROVED** [review]

**Pull request merged**